### PR TITLE
Removed the forgot password activity

### LINKED ISSUE #1326: Remove the current forgot password mechanism and make it simple
**Actual Behaviour**

Currently, whenever we forget the password, the forgot password option on LoginActivity opens up the ForgotPasswordActivity where we have to type the details once again and wait further, this is very fuzzy.

**Expected Behaviour**

On clicking the forgot password option instant confirmation should be shown.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1326 

Changes: Now when we click on the forgot password option on login screen instant confirmation is shown.

Screenshots for the change: 
screenshot_20180529-190948

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@batbrain7 Please fix your git config.

**Comment by batbrain7** [comment]
@iamareebjamal fixed it next commits will be shown in my name

File: app/build.gradle [hunk 1]
```diff
@@ -64,6 +64,7 @@ dependencies {
 
     //Leak Canary
     implementation 'com.android.support.constraint:constraint-layout:1.1.0'
+    implementation 'com.android.support:support-v4:27.1.1'
```
Comment (iamareebjamal) [inline app/build.gradle]: This is automatically pulled in transitively

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -51,11 +51,12 @@
...
             android:name=".login.LoginActivity"
             android:theme="@style/Login_Theme"
-            android:windowSoftInputMode="stateVisible|adjustResize"/>
+            android:windowSoftInputMode="stateVisible|adjustResize" />
         <activity
             android:name=".forgotpassword.ForgotPasswordActivity"
+            android:label="Reset Password"
             android:theme="@style/ForgetPass_Theme"
-            android:windowSoftInputMode="stateVisible|adjustResize" />
+            android:windowSoftInputMode="stateVisible|adjustResize"></activity>
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: Unnecessary changes

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -51,11 +51,12 @@
         <activity
             android:name=".login.LoginActivity"
             android:theme="@style/Login_Theme"
-            android:windowSoftInputMode="stateVisible|adjustResize"/>
+            android:windowSoftInputMode="stateVisible|adjustResize" />
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: UC

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -72,6 +73,15 @@
                 android:name="android.support.PARENT_ACTIVITY"
                 android:value="org.fossasia.susi.ai.skills.SkillsActivity" />
         </activity>
+        <activity
+            android:name=".login.ForgotPass"
+            android:configChanges="orientation|screenSize"
```
Comment (iamareebjamal) [inline app/src/main/AndroidManifest.xml]: Never lock orientation

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 5]
```diff
@@ -30,6 +30,7 @@ class LoginActivity : AppCompatActivity(), ILoginView {
 
     lateinit var loginPresenter: ILoginPresenter
     lateinit var progressDialog: ProgressDialog
+    lateinit var forgotPasswordProgressDialog: ProgressDialog
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: ProgressDialog is deprecated, use ProgressBar

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 6]
```diff
@@ -86,6 +95,7 @@ class LoginActivity : AppCompatActivity(), ILoginView {
             }
         }
         log_in.isEnabled = true
+        forgot_password.isEnabled = true
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Use camel case

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 7]
```diff
@@ -185,4 +186,36 @@ class LoginActivity : AppCompatActivity(), ILoginView {
         loginPresenter.onDetach()
         super.onDestroy()
     }
+
+    override fun resetPasswordSuccess(title: String?, message: String?) {
+        startActivity(Intent(this@LoginActivity, ForgotPass::class.java))
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: title and message are unneeded

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt [hunk 8]
```diff
@@ -165,4 +170,56 @@ class LoginPresenter(loginActivity: LoginActivity) : ILoginPresenter, ILoginMode
...
+            loginView?.resetPasswordFailure(utilModel.getString(R.string.email_invalid_title), utilModel.getString(R.string.email_invalid), utilModel.getString(R.string.retry), Color.RED)
+        } else {
+            loginView?.resetPasswordFailure("${response.code()} " + utilModel.getString(R.string.error), response.message(), utilModel.getString(R.string.ok), Color.BLUE)
+        }
+    }
+
+    override fun cancelSignup() {
+        forgotPasswordModel.cancelSignup()
+    }
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt]: The project is using MVP but all contributors seem to just put everything in the views. This is MVC guys, which is untestable even when using isolated components. And this is not even isolated. This is creating God Activities. Please refactor your code to follow SOLID and think about design as well. The goal of GSoC is to let you experience and become much better developers who can code beyond their comfort zone.

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 9]
```diff
@@ -185,4 +186,36 @@ class LoginActivity : AppCompatActivity(), ILoginView {
...
+    }
+
+    override fun resetPasswordFailure(title: String?, message: String?, button: String?, color: Int) {
+        val notSuccessAlertboxHelper = AlertboxHelper(this@LoginActivity, title, message, null, null, button, null, color)
+        notSuccessAlertboxHelper.showAlertBox()
+    }
+
+    override fun showForgotPasswordProgress(boolean: Boolean) {
+        if (boolean) forgotPasswordProgressDialog.show() else forgotPasswordProgressDialog.dismiss()
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: No

**Comment by iamareebjamal** [comment]
Good job. Just change the requested items

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt (hunk 6 above)
Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: @iamareebjamal should I make another issue for the camel case replacement in the app, because this one has multiple uses in the same activity and they have no relation to this PR.

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: OK, but I was referring to the new code. You should correct things one by one in at least new code. But yes, handle that in separate issue

Comment (batbrain7) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: the issue I created is #1330
